app/questionnaire/path_finder.py

```diff
@@ -34,31 +36,68 @@ def routing_path(
         """
         Visits all the blocks in a section and returns a path given a list of answers.
         """
-        blocks: List[Mapping] = []
-        routing_path_block_ids: List[str] = []
+        routing_path_block_ids: list[str] = []
         current_location = Location(section_id=section_id, list_item_id=list_item_id)
         section = self.schema.get_section(section_id)
         list_name = self.schema.get_repeating_list_for_section(
             current_location.section_id
         )
 
         if section:
+            when_rules_block_dependencies = self._get_when_rules_block_dependencies(
+                section["id"]
+            )
+            blocks = self._get_not_skipped_blocks_in_section(
+                current_location,
+                routing_path_block_ids,
+                section,
+                when_rules_block_dependencies,
+            )
+
+            if blocks:
+                routing_path_block_ids = self._build_routing_path_block_ids(
+                    blocks, current_location, when_rules_block_dependencies
+                )
+
+        return RoutingPath(routing_path_block_ids, section_id, list_item_id, list_name)
+
+    def _get_when_rules_block_dependencies(self, section_id: str) -> list[str]:
+        """NB: At present when rules block dependencies does not fully support repeating sections.
+        It is supported when the section is dependent i.e the current section is repeating and building the routing path for sections that are not,
+        It isn't supported if it needs to build the path for repeating sections"""
+        return [
+            block_id
+            for dependent_section in self.schema.when_rules_section_dependencies_map.get(
+                section_id, {}
+            )
+            for block_id in self.routing_path(dependent_section)
+            if dependent_section not in self.progress_store.started_section_keys()
```

Comment:
Section keys are tuple, so this should be:
```suggestion
            if (dependent_section, None) in self.progress_store.started_section_keys()
```

However, the fact that no tests fails indicates a gap in our unit test, so we need to have one for when the section has not started.

Reply:
There are tests for not started, but they don't test that the rule essentially evaluates false because dependent section block id are don't exist, hence the path should be different to if it was started.
> It was just all sections routing were still being evaluated

Correct, but that means that one test should have failed if we had that scenario covered. Coverage passes because it is a comprehension.

Reply:
As discussed, we can revisit this when we do iteration 2.